tools/docs/config.ts

```diff
@@ -159,22 +205,30 @@ export async function generateConfig(dist: string, bot = false): Promise<void> {
     '\n'
   );
 
+  const indexed = indexMarkdown(configOptionsRaw);
+
   options
     .filter((option) => option.releaseStatus !== 'unpublished')
```

Comment:
```suggestion
    .filter((option) => option.releaseStatus !== 'unpublished' && !!option.globalOnly === bot)
```

we should filter options, see discossion below

Reply:
> this wont work as `!!(undefined|false) == false` is a truthy, so all non self hosted cmds gets filter out.

This is what i expect. when `bot=false` we only want options where `globalOnly=false|undefined`.
In the other case we like to get only global options.

> 
> also, there are some (13) self hosted options that have `globalOnly` undefined e.g. redisUrl. can we also rely on `stage` maybe?

We need to fix those 13 options and add the globalOnly flag.
@rarkins This changes are probably breaking for some people, can we change them in a fix? Or do we need a major bump?

Reply:
- #16202